pkg/operator/controller/credentials-secret/credentials_secret.go

```diff
@@ -138,7 +146,14 @@ func (r *reconciler) updateCredentialsSecret(ctx context.Context, current, desir
 		return false, nil
 	}
 	updated := current.DeepCopy()
-	updated.Data = desired.Data
+	if desired.StringData != nil {
+		updated.StringData = make(map[string]string)
+		for key, value := range desired.StringData {
+			updated.StringData[key] = string(value)
+		}
+	} else {
+		updated.Data = desired.Data
+	}
```

Comment:
This not needed, `desired` variable already contains what's needed to be passed to `updated`.

Reply:
Removed